hermes/bin/src/ipfs/mod.rs

```diff
@@ -26,48 +29,69 @@ use crate::{
+/// `HermesIpfsNode` is first used. This is done to avoid any issues that may arise if
+/// the IPFS Node is initialized multiple times.
 ///
-/// The IPFS state is initialized in a separate thread and the sender channel is stored in
-/// the `HermesIpfsState`.
-static HERMES_IPFS_STATE: Lazy<HermesIpfsState> = Lazy::new(|| {
-    let sender = if let Ok(runtime) = Builder::new_current_thread().enable_all().build() {
-        let (sender, receiver) = mpsc::channel(1);
-        let _handle = std::thread::spawn(move || {
-            runtime.block_on(async move {
-                let h = tokio::spawn(ipfs_task(receiver));
-                drop(tokio::join!(h));
-            });
-            std::process::exit(0);
-        });
-        Some(sender)
-    } else {
-        // Failed to start the IPFS task
-        tracing::error!("Failed to start the IPFS task");
-        None
-    };
-    HermesIpfsState::new(sender)
+/// The IPFS Node is initialized in a separate thread and the sender channel is stored in
+/// the `HermesIpfsNode`.
+pub(crate) static HERMES_IPFS: Lazy<HermesIpfsNode> = Lazy::new(|| {
+    let hermes_home_dir = crate::cli::Cli::hermes_home().unwrap_or_else(|err| {
+        tracing::error!("Failed to get Hermes home directory: {}", err);
+        ".hermes".into()
+    });
+    let ipfs_data_path = hermes_home_dir.as_path().join("ipfs");
+    HermesIpfsNode::bootstrap(
+        IpfsBuilder::new()
+            .with_default()
+            .set_default_listener()
+            .disable_tls()
+            .set_disk_storage(ipfs_data_path.clone()),
+    )
+    .unwrap_or_else(|err| {
+        tracing::error!("Failed to bootstrap IPFS node: {}", err);
+        HermesIpfsNode::default()
+    })
 });
 
-/// Hermes IPFS Internal State
-struct HermesIpfsState {
-    /// State related to `HermesAppName`
+/// Hermes IPFS Internal Node
+pub(crate) struct HermesIpfsNode {
+    /// Send events to the IPFS node.
+    sender: Option<mpsc::Sender<IpfsCommand>>,
+    /// State related to `ApplicationName`
     apps: AppIpfsState,
 }
 
-impl HermesIpfsState {
-    /// Create a new `HermesIpfsState`
-    fn new(sender: Option<mpsc::Sender<IpfsCommand>>) -> Self {
-        Self {
-            apps: AppIpfsState::new(sender),
-        }
+impl HermesIpfsNode {
+    /// Create and bootstrap a new `HermesIpfsNode`
+    pub(crate) fn bootstrap(builder: IpfsBuilder) -> anyhow::Result<Self> {
+        tracing::info!("{} Bootstrapping IPFS node", console::Emoji::new("🖧", ""),);
+        let runtime = Builder::new_current_thread().enable_all().build()?;
+        let (sender, receiver) = mpsc::channel(1);
+        let _handle = std::thread::spawn(move || {
+            drop(runtime.block_on(async move {
+                // Build and start IPFS node
+                let node = builder.start().await?;
+                // Bootstrap the node with default addresses
+                let addrs = node.default_bootstrap().await?;
+                node.bootstrap().await?;
```

Comment:
Is it necessary to `bootstrap` node twice ?

Reply:
In this case, the first call adds the default public IPFS node addresses to the list, the second call actually connects to these nodes. I've put better comments around this code.